packages/@sanity/desk-tool/src/panes/documentPane/timeline/timeline.tsx

```diff
@@ -27,113 +22,91 @@ interface TimelineProps {
+        requestAnimationFrame(() => onLoadMore(isVisible))
       }
-    }, [onLoadMore, loadingElement])
-
-    // This is needed because we set the reference element both for
-    // the provided `ref` from `forwardRef`, and the local `rootRef`.
-    const setRef = useCallback(
-      (el: HTMLDivElement | null) => {
-        if (ref) {
-          if (typeof ref === 'function') ref(el)
-          if (typeof ref === 'object') (ref as any).current = el
-        }
-        rootRef.current = el
-      },
-      [ref]
-    )
-
-    // Load whenever it's needed
-    useEffect(checkIfLoadIsNeeded, [checkIfLoadIsNeeded])
-
-    // On mount: Scroll to selected timeline item
-    useEffect(() => {
-      const selectedEl = listRef.current?.querySelector('[data-state="selected"]')
-
-      if (selectedEl) {
-        window.requestAnimationFrame(() => {
-          selectedEl.scrollIntoView({block: 'center'})
-        })
-      }
-    }, [])
-
-    return (
-      <div className={styles.root} ref={setRef} onScroll={checkIfLoadIsNeeded}>
-        {timeline.chunkCount === 0 && (
-          <Stack padding={4} space={3} style={{maxWidth: 200}}>
-            <Text weight="bold">No document history</Text>
-            <Text muted size={1}>
-              When changing the content of the document, the document versions will appear in this
-              menu.
-            </Text>
-          </Stack>
-        )}
-        <ol className={styles.list} ref={listRef}>
-          {timeline.mapChunks((chunk) => {
-            const isSelectionTop = topSelection === chunk
-            const isSelectionBottom = bottomSelection === chunk
-
-            if (isSelectionTop) {
-              state = 'withinSelection'
-            }
-
-            if (isSelectionBottom) {
-              state = 'selected'
-            }
-
-            const item = (
-              <TimelineItem
-                chunk={chunk}
-                isSelectionBottom={isSelectionBottom}
-                isSelectionTop={isSelectionTop}
-                key={chunk.id}
-                state={state}
-                onSelect={onSelect}
-                type={chunk.type}
-                timestamp={chunk.endTimestamp}
-              />
-            )
-
-            // Flip it back to normal after we've rendered the active one.
-            if (state === 'selected') {
-              state = 'enabled'
-            }
-
-            return item
-          })}
-        </ol>
-
-        {!timeline.reachedEarliestEntry && (
-          <div className={styles.loading} ref={setLoadingElement}>
-            <Spinner center />
-          </div>
-        )}
-      </div>
-    )
-  }
-)
+    }
+  }, [onLoadMore, loadingElement])
+
+  // Load whenever it's needed
+  useEffect(checkIfLoadIsNeeded, [checkIfLoadIsNeeded])
+
+  return (
+    <Root ref={rootRef as any} onScroll={checkIfLoadIsNeeded} data-ui="timeline">
+      {timeline.chunkCount === 0 && (
+        <Stack padding={4} space={3} style={{maxWidth: 200}}>
```

Comment:
This inline `style` could be moved to a `const` outside the component + a `@todo` to refactor it to a styled-component later

Reply:
Good point! I actually moved it into a new styled component in the `timeline.styled` instead 👍